app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt

```diff
@@ -0,0 +1,307 @@
+        val viewModel = createViewModel(
+            awalaManager = awalaManager,
+            accountRepository = accountRepository,
+            contactsRepository = contactsRepository,
+            conversationsRepository = createConversationsRepository(
+                awalaManager = awalaManager,
+                contactsRepository = contactsRepository,
+                accountRepository = accountRepository,
+            ),
+        )
+        assertEquals(RootNavigationScreen.NoContactsScreen, viewModel.rootNavigationScreen.value)
+    }
+
+    @Test
+    fun `Test that Registration waiting screen is being opened, if Awala is successfully initialized, and account is not created yet (pending creation)`() {
+        val awalaManager = createAwalaManager(ioDispatcher = dispatcher)
+        val accountRepository = createAccountRepository(
+            accounts = listOf(createAccount(isCreated = false)),
+        )
+        val contactsRepository = createContactsRepository(
+            contacts = emptyList(),
+            isSentPairRequestOnce = false,
+            accountRepository = accountRepository,
+            awalaManager = awalaManager,
+        )
+        val viewModel = createViewModel(
+            awalaManager = awalaManager,
+            accountRepository = accountRepository,
+            contactsRepository = contactsRepository,
+            conversationsRepository = createConversationsRepository(
+                awalaManager = awalaManager,
+                contactsRepository = contactsRepository,
+                accountRepository = accountRepository,
+            ),
+        )
+        assertEquals(RootNavigationScreen.RegistrationWaiting, viewModel.rootNavigationScreen.value)
+    }
+
+    @Test
+    fun `Test that Home screen is being opened, if Awala is successfully initialized, contacts is empty, BUT there are some conversations`() {
+        val awalaManager = createAwalaManager(ioDispatcher = dispatcher)
+        val accountRepository = createAccountRepository(accounts = listOf(createAccount()), ioDispatcher = dispatcher)
+        val contactsRepository = createContactsRepository(
+            contacts = emptyList(),
+            isSentPairRequestOnce = true,
+            accountRepository = accountRepository,
+            awalaManager = awalaManager,
+            ioDispatcher = dispatcher,
+        )
+        val conversationId = UUID.randomUUID()
+        val viewModel = createViewModel(
+            awalaManager = awalaManager,
+            accountRepository = accountRepository,
+            contactsRepository = contactsRepository,
+            conversationsRepository = createConversationsRepository(
+                conversations = listOf(createConversation(conversationId)),
+                messages = listOf(createMessage(conversationId)),
+                awalaManager = awalaManager,
+                contactsRepository = contactsRepository,
+                accountRepository = accountRepository,
+            ),
```

Comment:
How about you call `createConversationsRepository()` inside `createViewModel()`? These tests have a lot of repetitive code, and I think this is an easy way to improve this.

However, if you implement my suggestion, you're going to have to add the optional parameters `conversations = emptyList()` and `messages = emptyList()` to `createViewModel()`.

Reply:
Sure, nice suggestion!